modules/sheet/editTags.js

```diff
@@ -0,0 +1,137 @@
+const concat = require('lodash/concat')
+const filter = require('lodash/filter')
+const forEach = require('lodash/forEach')
+const addSuggest = require('suggest-box')
+const TagHelper = require('scuttle-tag')
+
+exports.gives = nest('sheet.editTags')
+
+exports.needs = nest({
+  'about.obs.name': 'first',
+  'keys.sync.id': 'first',
+  'sbot.obs.connection': 'first',
+  'sheet.display': 'first',
+  'tag.html.tag': 'first',
+  'tag.obs.suggest': 'first'
+})
+
+exports.create = function (api) {
+  return nest({ 'sheet.editTags': editTags })
+
+  function editTags ({ msgId }, cb) {
+    const ScuttleTag = TagHelper(api.sbot.obs.connection)
+    const HtmlTag = api.tag.html.tag
+
+    cb = cb || function () {}
+
+    api.sheet.display(function (close) {
+      const { content, onMount, onSave } = edit({ msgId }, cb)
+
+      return {
+        content,
+        footer: [
+          h('button.save', { 'ev-click': publish }, 'Save'),
+          h('button.cancel', { 'ev-click': close }, 'Cancel')
+        ],
+        onMount
+      }
+
+      function publish () {
+        close()
+        onSave()
+      }
+    })
+
+    function edit ({ msgId }, cb) {
+      const tagsToCreate = MutantArray([])
+      const tagsToApply = MutantArray([])
+      const tagsToRemove = MutantArray([])
+      const tagsInput = Value('')
+
+      const myId = api.keys.sync.id()
+      const messageTags = map(ScuttleTag.obs.messageTagsFrom(msgId, myId), tagId =>
+        ScuttleTag.obs.Tag(tagId, api.about.obs.name))
+      const filteredMessages = computed([messageTags, tagsToRemove], (tags, removedIds) =>
+        filter(tags, tag => !removedIds.includes(tag.tagId)))
+
+      const messageTagsView = map(filteredMessages, tag =>
+        computed(tag, t => HtmlTag(t, () => tagsToRemove.push(t.tagId))))
+      const tagsToApplyView = map(tagsToApply, tag =>
+        HtmlTag(tag, () => tagsToApply.delete(tag)))
+      const tagsToCreateView = map(tagsToCreate, tag =>
+        HtmlTag({ tagName: tag, tagId: 'new' }, () => tagsToCreate.delete(tag)))
+      const stagedTags = computed([messageTagsView, tagsToApplyView, tagsToCreateView], (a, b, c) =>
+        h('StagedTags', concat(a, [b, c])))
+
+      const input = h('input.tags', {
+        placeholder: 'Add tags here',
+        'ev-keyup': onInput,
+        value: tagsInput()
+      })
+
+      input.addEventListener('suggestselect', onSuggestSelect)
+
+      return {
+        content: [stagedTags, h('EditTags', input)],
+        onMount,
+        onSave
+      }
+
+      function onMount () {
+        input.focus()
+        const appliedTagIds = map(filteredMessages, tag => tag.tagId)
+        const applyTagIds = map(tagsToApply, tag => tag.tagId)
+        const stagedTagIds = computed([ appliedTagIds, applyTagIds ], (a, b) => concat(a, b))
+        const getTagSuggestions = api.tag.obs.suggest(stagedTagIds)
+        addSuggest(input, (inputText, cb) => {
+          cb(null, getTagSuggestions(inputText))
```

Comment:
I've changed all my suggestions methods to be async, like so this line for me is : 

```
getTagSuggestions(inputText, cb)
```